data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -119,6 +120,17 @@ public GenesisData getGenesisData() {
         spec.atEpoch(ZERO).getConfig().getGenesisForkVersion());
   }
 
+  public tech.pegasys.teku.spec.datastructures.genesis.GenesisData getStateGenesisData() {
+    if (!isStoreAvailable()) {
+      throw new ChainDataUnavailableException();
+    }
+    return recentChainData.getGenesisData().orElseThrow(ChainDataUnavailableException::new);
+  }
```

Comment:
This does worry me, I'm not sure it's consistent with old handling (haven't checked), we'd want to make sure that a pre-genesis chain is not doing something horrible like throwing big errors... easiest way would be to set a minimal chain with a long pre-genesis and make sure the logs are clean when we call this... I'm starting to wonder if we should go back to using an object so that we can use `request.handleOptionalResult`

Reply:
A pre-genesis chain would return false from `isStoreAvailable()`. There's no way to have a Store without having the genesis set so this should be fine. It may in fact be worth removing the `isStoreAvailable` check and just using the `orElseThrow` to throw the `ChainDataUnavailableException` when genesis data is unavailable - currently it's essentially duplicating the check.

Reply:
This basically does the same as previously, except returns the actual data, rather than the wrapper class. @rolfyone